app/controllers/admin/order_cycles_controller.rb

```diff
@@ -43,51 +36,33 @@ def new
     end
 
     def create
-      @order_cycle = OrderCycle.new(params[:order_cycle])
+      @order_cycle_form = OrderCycleForm.new(@order_cycle, params, spree_current_user)
 
-      respond_to do |format|
-        if @order_cycle.save
-          OpenFoodNetwork::OrderCycleFormApplicator.new(@order_cycle, spree_current_user).go!
-          invoke_callbacks(:create, :after)
-          flash[:notice] = I18n.t(:order_cycles_create_notice)
-          format.html { redirect_to admin_order_cycles_path }
-          format.json { render :json => { success: true } }
-        else
-          format.html
-          format.json { render :json => { errors: @order_cycle.errors.full_messages }, status: :unprocessable_entity }
-        end
+      if @order_cycle_form.save
+        flash[:notice] = I18n.t(:order_cycles_create_notice)
+        render json: { success: true }
+      else
+        render json: { errors: @order_cycle.errors.full_messages }, status: :unprocessable_entity
       end
     end
 
     def update
-      @order_cycle = OrderCycle.find params[:id]
+      @order_cycle_form = OrderCycleForm.new(@order_cycle, params, spree_current_user)
 
-      respond_to do |format|
-        if @order_cycle.update_attributes(params[:order_cycle])
-          unless params[:order_cycle][:incoming_exchanges].nil? && params[:order_cycle][:outgoing_exchanges].nil?
-            # Only update apply exchange information if it is actually submmitted
-            OpenFoodNetwork::OrderCycleFormApplicator.new(@order_cycle, spree_current_user).go!
-          end
-          invoke_callbacks(:update, :after)
```

Comment:
what about these callbacks? they are not needed anymore?

Reply:
No, no longer required. They were added by me at the start of the subscriptions work. c2b68ed6b5ba2f033ee734c17aad2c669cce268a

These have been replaced by a method call in `OrderCycleForm#save`.